paddle/fluid/framework/ir/mkldnn/matmul_activation_mkldnn_fuse_pass.cc

```diff
@@ -0,0 +1,313 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/fluid/framework/ir/mkldnn/matmul_activation_mkldnn_fuse_pass.h"
+
+#include "paddle/fluid/framework/op_version_registry.h"
+#include "paddle/fluid/string/pretty_log.h"
+
+namespace paddle {
+namespace framework {
+namespace ir {
+
+using string::PrettyLogDetail;
+
+void MatmulActivationMkldnnFusePass::ApplyImpl(Graph* graph) const {
+  std::vector<std::string> act_types = {"abs",
+                                        "clip",
+                                        "gelu",
+                                        "hard_sigmoid",
+                                        "hard_swish",
+                                        "leaky_relu",
+                                        "mish",
+                                        "relu",
+                                        "relu6",
+                                        "sigmoid",
+                                        "sqrt",
+                                        "swish",
+                                        "tanh"};
+
+  std::vector<std::string> matmul_types = {"matmul"};
+
+  for (const auto& matmul_type : matmul_types)
+    for (auto& act_type : act_types) {
+      std::unordered_map<std::string, std::string> attrs_map;
+
+      if (act_type == "swish")
+        attrs_map.emplace("beta", "fuse_alpha");
+      else if (act_type == "relu6")
+        attrs_map.emplace("threshold", "fuse_alpha");
+      else if (act_type == "hard_sigmoid") {
+        attrs_map.emplace("slope", "fuse_alpha");
+        attrs_map.emplace("offset", "fuse_beta");
+      } else if (act_type == "clip") {
+        attrs_map.emplace("min", "fuse_alpha");
+        attrs_map.emplace("max", "fuse_beta");
+      } else {
+        attrs_map.emplace("alpha", "fuse_alpha");
+        attrs_map.emplace("beta", "fuse_beta");
+      }
+      FuseMatmulAct(graph, matmul_type, act_type, attrs_map);
+    }
+}
+
+void MatmulActivationMkldnnFusePass::FuseMatmulAct(
+    Graph* graph,
+    const std::string& matmul_type,
+    std::string& act_type,
+    const std::unordered_map<std::string, std::string>& attrs_map) const {
+  PADDLE_ENFORCE_NOT_NULL(
+      graph, platform::errors::InvalidArgument("Graph cannot be nullptr."));
+  FusePassBase::Init(matmul_type + "_" + act_type + "_mkldnn_fuse_pass", graph);
+
+  GraphPatternDetector gpd;
+  patterns::OperatorActivation matmul_act_pattern(
+      gpd.mutable_pattern(), "matmul_activation_mkldnn_fuse");
+  matmul_act_pattern(matmul_type, act_type);
+
+  int found_matmul_activation_count = 0;
+  auto handler = [&](const GraphPatternDetector::subgraph_t& subgraph,
+                     Graph* g) {
+    VLOG(4) << "handle " + matmul_type + "+" + act_type + " fuse";
+
+    if (!IsCompat(subgraph, g)) {
+      LOG(WARNING) << "matmul_activation_mkldnn_fuse_pass op compat failed.";
+      return;
+    }
```

Comment:
Do we need this if here?
It can disable this fuse if this method will complain about custom attrs etc.

Reply:
I think this topic needs further discussion, because some fuse passes have `OpCompat` for each fusion combination and others don't have any. We should decide if it gives us any benefit and then unify all passes